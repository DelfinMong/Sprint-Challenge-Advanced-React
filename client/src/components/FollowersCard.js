import React from "react";

const FollowersCard = props => {
  return (
    <div>
      <h1>Followers</h1>
      <p>{props.login}</p>
      <img src={props.img} alt="avatar" />
    </div>
  );
};

export default FollowersCard;