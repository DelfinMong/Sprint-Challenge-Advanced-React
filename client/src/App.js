import React from 'react';
import axios from  'axios'
import Usercard from "./components/Usecard";
import FollowersCard from "./components/FollowersCard";
import DarkMode from './components/DarkMode'
import './App.css';

class App extends React.Component {
  
  state = {
    followersList: [],
  };

  componentDidMount() {
    axios
      .get("https://api.github.com/users/DelfinMong/followers")
      .then(res => {
        console.log("Followers API: ", res);
        this.setState({
          followersList: res.data,
        });
      })
      .catch(err => console.log("have an error", err));
  }

  render() {
    return (
      <div className="App">
        <Usercard login={this.state.login} img={this.state.img} />
         <DarkMode />
        {this.state.followersList.map(item => {
          return <FollowersCard login={item.login} img={item.avatar_url} />;
        })}
      </div>
    );
  }

}

export default App;

// function App() {
//   return (
//     <div className="App">
//       <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />
//         <p>
//           Edit <code>src/App.js</code> and save to reload.
//         </p>
//         <a
//           className="App-link"
//           href="https://reactjs.org"
//           target="_blank"
//           rel="noopener noreferrer"
//         >
//           Learn React
//         </a>
//       </header>
//     </div>
//   );
// }


