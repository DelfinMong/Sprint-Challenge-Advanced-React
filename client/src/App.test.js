import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {render} from '@testing-library/react';
import FollowersCard from './components/FollowersCard';
import renderer from 'react-test-renderer';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

test(' rtl renders w/o crashing', () => {
  render(<App />);
});

test('searching for card and follower', () => {
  const { getByText } = render(<App />);
  getByText(/card/i);
  // getByText(/Followers/i);
});

test('renders correctly when there are no items', () => {
  const tree = renderer.create(<FollowersCard />).toJSON();
  expect(tree).toMatchSnapshot();
});



