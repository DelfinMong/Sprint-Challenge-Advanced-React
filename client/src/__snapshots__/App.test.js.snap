// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <h1>
    Followers
  </h1>
  <p />
  <img
    alt="avatar"
  />
</div>
`;
